### ISSUE TITLE
Use published as default status

### ISSUE THREAD

**Issue by giomasce** [getpelican/pelican#2558]
Hi,

at https://github.com/getpelican/pelican/blob/master/pelican/contents.py#L144 the `status` of a content is set to `None` is `default_status` is not specified. This deterministically causes an error as soon as `lower` is called on it at https://github.com/getpelican/pelican/blob/master/pelican/contents.py#L426. Wouldn't it more sensible to take `published` as default status?

Thanks for Pelican, it's really nice.

**Comment by giomasce** [comment]
BTW, I found the bug in action here: https://github.com/bstpierre/pelican-comments/pull/4

**Comment by avaris** [comment]
Yes and no. `Content` isn't meant to be used by itself. It's up to the [subclass to define that](https://github.com/getpelican/pelican/blob/master/pelican/contents.py#L505).

**Comment by giomasce** [comment]
Ok. In that case I suggest to error our as soon as no `status` and no `default_status` is found, so that the cause of the error is clear.

**Comment by avaris** [comment]
Actually, looking at it again, `getattr` fall back being `None` doesn't make much sense. `''` or `'published'` is probably OK:

```python
self.status = getattr(self, 'default_status', '')
```

Or, `setter` can deal with that.

**Comment by trevorld** [comment]
I can conifrm that 

````
self.status = getattr(self, 'default_status', '')
````

or

````
self.status = getattr(self, 'default_status', 'published')
````

or

````
    @status.setter
    def status(self, value):
        if value is None:
            self._status = value.lower()
        else:
            self._status = None
````

all get the "pelican comment system" plugin working again.

**Comment by justinmayer** [comment]
Who would be so kind as to submit a pull request that changes the default status from `None` to `''`? 😁

**Comment by MinchinWeb** [comment]
Pull Request created!

**Comment by justinmayer** [comment]
Thanks again for the fix, @MinchinWeb. The fix for this issue has been automatically published as [Pelican 4.1.1](https://github.com/getpelican/pelican/releases/tag/4.1.1), courtesy of [AutoPub](https://github.com/autopub/autopub). ✨
